src/CMakeLists.txt

```diff
@@ -129,7 +129,9 @@ set(MARIAN_GIT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../.git)
 if(NOT IS_DIRECTORY ${MARIAN_GIT_DIR}) # i.e., it's a submodule
   file(READ ${MARIAN_GIT_DIR} MARIAN_GIT_DIR)
   string(REGEX REPLACE "gitdir: (.*)\n" "\\1" MARIAN_GIT_DIR ${MARIAN_GIT_DIR})
-  get_filename_component(MARIAN_GIT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../${MARIAN_GIT_DIR}" ABSOLUTE)
+  if(NOT IS_ABSOLUTE ${MARIAN_GIT_DIR})
+    get_filename_component(MARIAN_GIT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../${MARIAN_GIT_DIR}" ABSOLUTE)
```

Comment:
A minor question: why are we making sure that it is an absolute path? Is it required to work in the upper module or safer? Add a comment if you think it is worth it.

Reply:
Notice the NOT. It's a bit complicated, but for submodules, git keeps the git information in the root directory and stores the path to the relevant directory in a file named .git in the submodule. The path can be either relative or absolute, as I found out the hard way. I can't remember all the details, but it may have also involved running things in host-mounted directories in compiling in Docker containers for dockerization. What I do remember is that at some point, things didn't work and that fixed it. For all practical purposes, is has no effect on run-of-the-mill marian compilation, because then .git is always a directory.

Reply:
Thanks!